src/components/CDPDisplay/index.js

```diff
@@ -8,25 +8,45 @@ import useStore from 'hooks/useStore';
 import { getCdp } from 'reducers/cdps';
 import { trackCdpById } from 'reducers/multicall/cdps';
 import CDPViewPresentation from './Presentation';
+import Unavailable from '../Unavailable';
+import { Routes, ZERO_ADDRESS } from '../../utils/constants';
+import { useNavigation } from 'react-navi';
 
 function CDPView({ cdpId }) {
   cdpId = parseInt(cdpId, 10);
   const { maker, account, network } = useMaker();
   const { show: showSidebar } = useSidebar();
   const [{ cdps, feeds }, dispatch] = useStore();
   const [cdpOwner, setOwner] = useState();
+  const [cdpAvailable, setCdpAvailable] = useState(true);
+  const navigation = useNavigation();
 
   useEffect(() => {
     (async () => {
+      async function redirect(account) {
+        const { search } = (await navigation.getRoute()).url;
+        navigation.navigate({
+          pathname: `/${Routes.BORROW}/owner/${account.address}`,
+          search
+        });
+      }
       const proxyAddress = await maker
         .service('mcd:cdpManager')
         .getOwner(cdpId);
-      const cdpOwnerAddress = await maker
-        .service('proxy')
-        .getOwner(proxyAddress);
-      setOwner(cdpOwnerAddress);
+      if (proxyAddress === ZERO_ADDRESS) {
+        account ? redirect(account) : setCdpAvailable(false);
+      } else {
+        try {
+          const cdpOwnerAddress = await maker
+            .service('proxy')
+            .getOwner(proxyAddress);
+          setOwner(cdpOwnerAddress);
+        } catch (err) {
+          setOwner(proxyAddress);
```

Comment:
@levity I think this should handle all 3 situations now:

1. if `proxyAddress === ZERO_ADDRESS`, vault doesn't exist, move on.
2. if `getOwner` throws, catch the error & use `proxyAddress` since we know it's a valid vault ID.
3. use the `cdpOwnerAddress`.

Let me know if there's one I missed.

Reply:
Hmm. `getOwner` should really return some sentinel value (e.g. `0x0`) rather than throwing if the address passed in is not a proxy. But anyway, this looks good to me.

Reply:
can you add a comment to that effect here? like `// cdp is not owned by a proxy`